src/Searching.ts

```diff
@@ -248,7 +248,7 @@ async function localPagination(
 
     // We only need to restore the encryption state for the new results, so
     // remember how many of them we got.
-    const newResultCount = localResult.results.length;
+    const newResultCount = localResult.results?.length ?? 0;
```

Comment:
Trying to understand what's going on here. This looks to be dealing with seshat results rather than `/search` response? Is that correct, or am I missing something?

According to the types, `results` is always present, so are the types wrong?

Reply:
> This looks to be dealing with seshat results rather than /search response? Is that correct, or am I missing something?

You are correct, but seshat mimics the C-S API so we should assume the same lack of required fields.

> According to the types, results is always present, so are the types wrong?

Yes, those are handled by https://github.com/matrix-org/matrix-js-sdk/pull/4228